File: Python Katas/Kata_02/solutions.py
# todo.py

def calculate_average(numbers):
    """
    Calculate and return the average of a list of numbers.
    """

    return sum(numbers)/len(numbers)


def is_palindrome(s):
    """
    Check if a given string is a palindrome (ignoring spaces and case).
    """
    s = "".join(s.split())
    return s.lower() == s[::-1].lower()


def flatten_list(nested_list):
    """
    Flatten a nested list into a single list.
    """
    flattened = []
    
    def recursive_flatten(item):
        if isinstance(item, list):
            for subitem in item:
                recursive_flatten(subitem)
        else:
            flattened.append(item)
    
    recursive_flatten(nested_list)
    return flattened
                      


def word_count(text):
    """
    Count the occurrences of each word in a given text and return a dictionary.
    """
    text_as_list = text.split()
    
    return {k:text_as_list.count(k) for k in text_as_list}

print(word_count(""))


def find_common_elements(list1, list2):
    """
    Find and return the common elements between two lists.
    """
    pass

File: JavaScript/Kata_03/solutions.js
/**
 * @param {Number} a
 * @param {Number} b
 * @returns {Number}
 */
function addTwoNumbers(a,b) {
    return a+b;
}

/**
 * 
 * @param {String} s 
 * @returns {String}
 */
function reverseString(s) {
    return s.split("").reverse().join("");
}

/**
 * @param {Array}
 * @returns {Number}
 */
function calculateAverage(numbers) {
    return numbers.reduce((a,b) => a+b ,0) / numbers.length;
}

/**
 * @param {Array}
 * @returns {Array}
 */
function flattenList(array) {
    return array.flat(Infinity);
}

/**
 * @param {String} text
 * @returns {Map}
 */
function wordCount(text) {
    let numOfWords = (list,e) => list.filter((a) => a === e).length;
    return text.split(" ").reduce((acc, curr) => 
        acc.set(curr, numOfWords(text.split(" "),curr)),new Map());
}

/**
 * @param {Array} list1
 * @param {Array} list2
 * @returns {Array}
 */
function findCommonElements(list1, list2) {
    return list1.filter((x) => list2.includes(x));
}

